app/src/main/java/com/jnj/vaccinetracker/visit/zscore/NutritionZScoreCalculator.kt

```diff
@@ -0,0 +1,58 @@
+package com.jnj.vaccinetracker.visit.zscore
+
+import android.graphics.Color
+import com.jnj.vaccinetracker.R
+import com.jnj.vaccinetracker.common.domain.entities.Gender
+import okhttp3.internal.toHexString
+import kotlin.random.Random
+
+class NutritionZScoreCalculator(
+        private val weight: Int?,
+        private val height: Int?,
+        private val isOedema: Boolean?,
+        gender: Gender,
+        birtDayText: String
+): ZScoreCalculator(gender, birtDayText) {
+   companion object {
+      const val NORMAL = "Normal Nutrition Status" // Green
+      const val MODERATE = "Moderate Acute Malnutrition" // Yellow
+      const val SEVERE_WITH_OEDEMA = "Severe Acute Malnutrition with Oedema" // Red
+      const val SEVERE_WITHOUT_OEDEMA = "Severe Acute Malnutrition without Oedema" // Red
+   }
+   override fun calculateZScoreAndRating(): ZScoreAndRating? {
+      val zScore = this.zScore ?: return null
+      val rating = when {
+         (zScore <= -3) && isOedema == true -> SEVERE_WITH_OEDEMA
+         zScore <= -3 -> SEVERE_WITHOUT_OEDEMA
+         zScore < -2.0 -> MODERATE
+         else -> NORMAL
+      }
+      return ZScoreAndRating(zScore, rating)
+   }
+
+   override fun calculateZScore(): Double? {
+      // Placeholder calculation (random value for demonstration)
+      // Perform calculations based on height, gender, and age to calculate Z-score
+      // Replace the placeholder calculation with actual logic using reference data or models
+      if (height == null || weight == null) return null
+      if (weight < 30) return -3.0
+      return Random.nextDouble(-2.9, 4.0)
+   }
+
+   fun isOedemaValue(): Boolean {
+      val zScore = this.zScore ?: return false
+      if (zScore <= -3) return true
+      return false
+   }
+
+   fun getTextColorBasedOnZsCoreValue(): Int {
+      val defaultColor = R.color.colorTextOnLight
+      val zScore = this.zScore ?: return defaultColor
+      val color = when {
+         zScore <= -3 -> Color.RED
+         zScore < -2.0 -> Color.YELLOW
```

Comment:
Lets change this to something **black** or different color ,   Yellow isnt good.  Check out 
![Screenshot 2024-06-13 070048](https://github.com/johnsonandjohnson/vxnaid/assets/48015736/90c2336a-d188-400d-851b-e1332d7a1f99)

Reply:
that was a requirement, however i can change it to something darker

Reply:
Its not readable as you can see. Thanks